designs/disruption-controls-by-reason.md

```diff
@@ -0,0 +1,271 @@
+# Disruption Controls By Reason 
+## User Scenarios 
+1. Users need the capability to schedule upgrades only during business hours or within more restricted time windows. Additionally, they require a system that doesn't compromise the cost savings from consolidation when upgrades are blocked due to drift.
+2. High-Frequency Trading (HFT) firms require full compute capacity during specific operational hours, making it imperative to avoid scale-down requests for consolidation during these periods. However, outside these hours, scale-downs are acceptable.
+See Less Made Up Scenarios here: 
+- https://github.com/kubernetes-sigs/karpenter/issues/924 
+- https://github.com/kubernetes-sigs/karpenter/issues/753#issuecomment-1790110838
+- https://github.com/kubernetes-sigs/karpenter/issues/672
+
+## Known Requirements 
+**Reason and Budget Definition:** Users should be able to define an reason and a corresponding budget(s).
+**Supported Reasons:** All disruption Reasons affected by the current Budgets implementation (Consolidation, Emptiness, Expiration, Drift) should be supported. 
+**Default Behavior for Unspecified Reasons:** Budgets should continue to support a default behavior for all disruption reasons. If an reason is unspecified, it is assumed to apply to all reasons. If a reason is unspecified, we apply the budget to be shared by all reasons that are unspecified. 
```

Comment:
nit: this section is a bit hard to parse.

> If an reason is unspecified, it is assumed to apply to all reasons. If a reason is unspecified, we apply the budget to be shared by all reasons that are unspecified.

becomes

> If a reason is unspecified:
>* it is assumed to apply to all reasons
>* we apply the budget to be shared by all reasons that are unspecified

*Or alternatively*
> If an reason is unspecified, it is assumed to apply to all reasons. If a reason is unspecified, we apply the budget to be shared by all reasons that are unspecified.

becomes
> If an reason is unspecified, it is assumed to apply to all reasons. We will apply the budget to be shared by all reasons that are unspecified.

Reply:
yeah good catch